python/ray/train/_internal/backend_executor.py

```diff
@@ -521,13 +523,14 @@ def _restart(self):
     def _increment_failures(self):
         self._num_failures += 1
         if self._num_failures >= self._max_failures:
+            _ray_start_tb = True  # noqa: F841
```

Comment:
this implementation feels a little bit like too much magic. Is there another way we can shorten tracebacks without leaving this magical variable everywhere?